### PR TITLE
Adding extension-less Office file snippet

### PR DESCRIPTION
**Pull request by aidenmitchell**
Office files without an extension appear as
```
.file_extension is null
and .file_type == "unknown"
and .content_type == "application/octet-stream"
```
in MQL, so adding this snippet to all of our Office file rules to ensure coverage. Size check is a guard rail.

### CODE REVIEW

File: detection-rules/attachment_cve_2023_21716_rtf_fonts.yml [hunk 1]
```diff
@@ -15,6 +15,12 @@ source: |
             .file_extension in~ ("rtf", "doc", "docx")
             or .file_extension in~ $file_extensions_common_archives
             or .file_extension in~ $file_extensions_macros
+            or (
+                .file_extension is null
+                and .file_type == "unknown"
```
Comment (rw-access) [inline detection-rules/attachment_cve_2023_21716_rtf_fonts.yml:20]: Not blocking you, but it sounds like we should add more file type matchers based on the contents? That way we can detect pdf(x), ppt(x), doc(x), etc

Comment (rw-access) [reply detection-rules/attachment_cve_2023_21716_rtf_fonts.yml:20]: totally, nothing you need to change here. Just making an observation that it sounds like we're short on some file type matchers in the backend

**Comment by morriscode** [comment]
100000 is only .1mb. might want to bump it?

**Review by morriscode — APPROVED** [review]

**Pull request merged**